### PR TITLE
Bugfix/store

### PR DESCRIPTION
**Pull request by michaloo**
- implements new metric
- saves the Hubspot unique ID - `vid`
- removes `fetched_at` field

### CODE REVIEW

**Comment by michaloo** [comment]
[Add hubspot contact id as hubspot/id trait](https://trello.com/c/1uSNBWBN/126-add-hubspot-contact-id-as-hubspot-id-trait)

**Comment by michaloo** [comment]
After playing with datadog dashboard I see the need of tagging environments.
Devs would like to instrument also non-production envs to test and monitor their behaviour, yet the default for more business side user should be always and only production.

I thought we can stick with `host` information, but it can give some unnecessary noise to the dashboard settings. @sbellity

File: server/lib/mapping.js [hunk 1]
```diff
@@ -87,4 +87,22 @@ export default class Mapping {
...
+   */
+  getIdentFromHubspot(hubspotUser) {
+    const ident = {};
+
+    if (_.get(hubspotUser, "properties.email.value")) {
+      ident.email = _.get(hubspotUser, "properties.email.value");
+    }
+
+    if (hubspotUser.vid) {
+      ident.anonymous_id = `hubspot:${hubspotUser.vid}`;
```
Comment (michaloo) [inline server/lib/mapping.js]: @unity This is the place we set the anonymous id.

File: server/lib/mapping.js [hunk 2]
```diff
@@ -46,7 +46,7 @@ export default class Mapping {
       return traits;
     }, {});
 
-    hullTraits["hubspot/fetched_at"] = new Date();
+    hullTraits["hubspot/vid"] = userData.vid;
```
Comment (michaloo) [inline server/lib/mapping.js]: @unity @sbellity is it too much like hardcoding that mapping?
What do you think?

Comment (sbellity) [inline server/lib/mapping.js]: Let's try to be consistent and call that `hubspot/id`
+1 for hardcoding it

File: server/lib/hubspot-agent.js [hunk 3]
```diff
@@ -182,4 +183,36 @@ export default class HubspotAgent {
...
+  * It tries to get the data from user's information, if not available
+  * defaults to one hour from now.
+  *
+  * @return {Promise -> lastImportTime (ISO 8601)} 2016-08-04T12:51:46Z
+  */
+  getLastUpdate() {
+    const defaultValue = moment().subtract(1, "hour").format();
+
+    return this.hullClient.get("/search/user_reports", {
+      include: ["traits_hubspot/updated_at"],
```
Comment (sbellity) [inline server/lib/hubspot-agent.js]: @michaloo where is this value actually set ?
I see that you have removed the `hubspot/fetched_at` trait but I do not see where it's replaced by `hubspot/updated_at`.

Comment (michaloo) [inline server/lib/hubspot-agent.js]: Good question. The `hubspot/updated_at` trait is defined here in the mapping:
https://github.com/hull-ships/hull-hubspot/blob/develop/server/lib/mapping-data.js#L24

and if you look at the method which builds the object for traits call, it takes all fields from default mapping, optionally adding stuff from settings:
https://github.com/hull-ships/hull-hubspot/blob/develop/server/lib/mapping-data.js#L63

I've decided to remove `hubspot/fetched_at` (as set every time to `new Date()`) to limit the  changes which happen to user object. We had similar problem on mailchimp ship previously.

Comment (sbellity) [inline server/lib/hubspot-agent.js]: Alright, thanks I missed that one.

**Pull request merged**